platform/core/src/core.rs

```diff
@@ -117,7 +126,49 @@ pub fn load_feature_categories(
     )
 }
 
-pub fn exec_rdf_query(query: SPARQLQuery) -> Result<String, CoreFailure> {
+fn open_store(fs_root: String, active_feature_id: String) -> Result<Store, CoreFailure> {
+    Store::open(Path::new(&format!("{}/{}/{}", fs_root, active_feature_id, env!("RDF_DB_PATH")))).map_err(CoreFailure::map_storage_error)
+}
+
+pub fn open_feature_rdf_store() -> Result<(), CoreFailure> {
+    let mut instance = get_instance()?;
+    if instance.active_feature_id == None {
+        return Err(CoreFailure::no_active_feature("Open feature rdf store".to_string()))
+    }
+    match &instance.active_feature_id {
+        Some(id) => match open_store(instance.fs_root.to_string(), id.to_string()) {
+            Ok(store) => {
+                instance.feature_rdf_store = Some(store);
+                Ok(())
+            },
+            Err(error) => Err(error)
+        },
+        _ => Err(CoreFailure::no_active_feature("Open feature rdf store".to_string()))
+    }
+}
+
+pub fn exec_feature_rdf_query (
+    query: SPARQLQuery,
+) -> Result<QueryResults, CoreFailure> {
+    let instance = get_instance()?;
+    match &instance.feature_rdf_store {
+        Some(store) => store.query(&query).map_err(CoreFailure::map_sparql_evaluation_error),
+        _ => Err(CoreFailure::feature_store_not_initialized()) 
```

Comment:
Isn't it possible that there are other kind of errors? For instance, `query` being an update query.

Reply:
query returns Result<QueryResults, EvaluationError> by typing